### PR TITLE
[JENKINS-73259] Migrate Winstone from EE 8 to EE 9

### PR DESCRIPTION
**Pull request by basil**
### Context

Goes with three other pull requests:

- https://github.com/jenkinsci/jelly/pull/114
- https://github.com/jenkinsci/stapler/pull/482
- https://github.com/jenkinsci/jenkins/pull/9672

### Description

Replace dependencies with EE 9 versions in `pom.xml` and update all relevant imports.

### Testing done

Full PCT/ATH

### CODE REVIEW

**Comment by pokarand** [comment]
Jenkin

**Review by timja — APPROVED** [review]
nice work!

**Review by alecharp — APPROVED** [review]

File: src/main/java/winstone/Launcher.java [hunk 1]
```diff
@@ -183,7 +182,7 @@ public Launcher(Map<String, String> args) throws IOException {
             }
 
             // JENKINS-73616: Turn down log level of annotation parser
-            java.util.logging.Logger logger = java.util.logging.Logger.getLogger(AnnotationParser.class.getName());
+            java.util.logging.Logger logger = java.util.logging.Logger.getLogger("org.eclipse.jetty.ee9.annotations.AnnotationParser");
```
Comment (basil) [inline src/main/java/winstone/Launcher.java:185]: This annotation is no longer on our classpath in EE 9 due to some upstream cleanup in the Jetty dependency tree, so just use a string constant.

**Review by MarkEWaite — APPROVED** [review]

**Review by Kevin-CB — APPROVED** [review]
Looks fine, security wise.

**Review by BorisYaoA — APPROVED** [review]
LGTM

**Review by daniel-beck — APPROVED** [review]

**Pull request merged**